router/middleware/perm/perm.go

```diff
@@ -100,6 +100,15 @@ func MustWorkerAuthToken() gin.HandlerFunc {
 			"worker": cl.Subject,
 		}).Debugf("verifying worker %s has a valid auth token", cl.Subject)
 
+		// global permissions bypass
+		if cl.IsAdmin {
+			logrus.WithFields(logrus.Fields{
+				"user": cl.Subject,
+			}).Debugf("user %s has platform admin permissions", cl.Subject)
+
+			return
+		}
+
```

Comment:
With the introduction of this platform admin bypass, I wonder if we should restrict the fields that they can change (such as running build IDs)?

Reply:
@ecrupper can you explain a little bit more on this?

Reply:
I'm imagining a future case where a platform admin may want to live-update the worker's routes. In that case, because of [this check](https://github.com/go-vela/server/pull/772/files#diff-c760a1133ea01608d485599741679f949ff6f09ff14ba119fb15f67d81fecde4R101-R104) (which will return as `true` because `GetRunningBuildIDs` returns `[]string{}` if it is `nil` rather than `nil`), then the platform admin will also have to know the running build IDs to make this update.

Steps to reproduce:
1. fire up local stack on this branch
2. set user to platform admin
3. manually perform DB update to worker, setting running build ids to an example value
4. call PUT worker as platform admin with the body as: 
```json
{
	"id": 1,
	"hostname": "worker",
	"address": "http://worker:8080",
	"routes": [
		"vela",
		"highway"
	]
}
```
5. worker in DB will now have a running_build_ids value of empty. This is with the `latest` worker version, so no other updates have been made behind the scenes.

So either the platform admin shouldn't even be allowed to update values such as `running_build_ids`, or there should be a more scrutinizing check in the PUT call, which will not accidentally set values on behalf of the platform admin. I have no opinion on which of those two solutions is better, or if there's another option.

Reply:
Addressed by not requiring a running build list in the body.